dotnet/src/Connectors/Connectors.Memory.Redis/RedisHashSetVectorStoreRecordMapper.cs

```diff
@@ -0,0 +1,170 @@
+using System.Text.Json.Nodes;
+using Microsoft.SemanticKernel.Data;
+using StackExchange.Redis;
+
+namespace Microsoft.SemanticKernel.Connectors.Redis;
+
+/// <summary>
+/// Class for mapping between a hashset stored in redis, and the consumer data model.
+/// </summary>
+/// <typeparam name="TConsumerDataModel">The consumer data model to map to or from.</typeparam>
+internal sealed class RedisHashSetVectorStoreRecordMapper<TConsumerDataModel> : IVectorStoreRecordMapper<TConsumerDataModel, (string Key, HashEntry[] HashEntries)>
+    where TConsumerDataModel : class
+{
+    /// <summary>A property info object that points at the key property for the current model, allowing easy reading and writing of this property.</summary>
+    private readonly PropertyInfo _keyPropertyInfo;
+
+    /// <summary>The name of the temporary json property that the key field will be serialized / parsed from.</summary>
+    private readonly string _keyFieldJsonPropertyName;
+
+    /// <summary>A list of property info objects that point at the data properties in the current model, and allows easy reading and writing of these properties.</summary>
+    private readonly IEnumerable<PropertyInfo> _dataPropertiesInfo;
+
+    /// <summary>A list of property info objects that point at the vector properties in the current model, and allows easy reading and writing of these properties.</summary>
+    private readonly IEnumerable<PropertyInfo> _vectorPropertiesInfo;
+
+    /// <summary>A dictionary that maps from a property name to the configured name that should be used when storing it.</summary>
+    private readonly Dictionary<string, string> _storagePropertyNames;
+
+    /// <summary>A dictionary that maps from a property name to the configured name that should be used when serializing it to json for data and vector properties.</summary>
+    private readonly Dictionary<string, string> _jsonPropertyNames = new();
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="RedisHashSetVectorStoreRecordMapper{TConsumerDataModel}"/> class.
+    /// </summary>
+    /// <param name="keyPropertyInfo">The property info object that points at the key property for the current model.</param>
+    /// <param name="dataPropertiesInfo">The property info objects that point at the payload properties in the current model.</param>
+    /// <param name="vectorPropertiesInfo">The property info objects that point at the vector properties in the current model.</param>
+    /// <param name="storagePropertyNames">A dictionary that maps from a property name to the configured name that should be used when storing it.</param>
+    public RedisHashSetVectorStoreRecordMapper(
+        PropertyInfo keyPropertyInfo,
+        IEnumerable<PropertyInfo> dataPropertiesInfo,
+        IEnumerable<PropertyInfo> vectorPropertiesInfo,
+        Dictionary<string, string> storagePropertyNames)
+    {
+        Verify.NotNull(keyPropertyInfo);
+        Verify.NotNull(dataPropertiesInfo);
```

Comment:
We should do the checking for cases where someone is using an old version of c#, that doesn't have the compile time null checking, or in case someone just suppressed the warning/error.  That said, these are internal, so not much risk of that here.  I have this in a lot of places, so I can do a separate pass to remove them, if you feel strongly about removing it.

No need to check for empty here, since these are not strings.